worker/pkg/workflows/datasync/activities/activities.go

```diff
@@ -563,32 +563,61 @@ func getMysqlDsn(
 	}
 }
 
-func (b *benthosBuilder) buildProcessorMutation(ctx context.Context, cols []*mgmtv1alpha1.JobMapping) (string, error) {
-	pieces := []string{}
+func (b *benthosBuilder) buildProcessorConfig(ctx context.Context, cols []*mgmtv1alpha1.JobMapping) (*neosync_benthos.ProcessorConfig, error) {
+	mutations := []string{}
+	javascript := []string{}
 
 	for _, col := range cols {
-		if col.Transformer != nil && col.Transformer.Source != "" && col.Transformer.Source != "passthrough" && col.Transformer.Source != "generate_default" {
+		if shouldProcessColumn(col.Transformer) {
 
 			if _, ok := col.Transformer.Config.Config.(*mgmtv1alpha1.TransformerConfig_UserDefinedTransformerConfig); ok {
 
 				// handle user defined transformer -> get the user defined transformer configs using the id
 
 				val, err := b.convertUserDefinedFunctionConfig(ctx, col.Transformer)
 				if err != nil {
-					return "", errors.New("unable to look up user defined transformer config by id")
+					return nil, errors.New("unable to look up user defined transformer config by id")
 				}
 				col.Transformer = val
 
 			}
 
-			mutation, err := computeMutationFunction(col)
-			if err != nil {
-				return "", fmt.Errorf("%s is not a supported transformer: %w", col.Transformer, err)
+			if col.Transformer.Source == "javascript" {
+				// parse the js code and replace the key param word which is 'value' with the column name
+				parsed := parseJavascriptForColumnName(col.Transformer.Config.GetJavascriptConfig().GetCode(), "value", col.Column)
+				javascript = append(javascript, parsed)
+
+			} else {
+
+				mutation, err := computeMutationFunction(col)
+				if err != nil {
+					return nil, fmt.Errorf("%s is not a supported transformer: %w", col.Transformer, err)
+				}
+				mutations = append(mutations, fmt.Sprintf("root.%s = %s", col.Column, mutation))
 			}
-			pieces = append(pieces, fmt.Sprintf("root.%s = %s", col.Column, mutation))
+
 		}
 	}
-	return strings.Join(pieces, "\n"), nil
+
+	pc := &neosync_benthos.ProcessorConfig{
+		Mutation: strings.Join(mutations, "\n"),
+		Javascript: neosync_benthos.JavascriptConfig{
+			Code: strings.Join(javascript, "\n"),
+		},
```

Comment:
This should be conditional here based on the presence of these configs.

So essentially something like this:

```
mutation := strings.Join(mutations, "\n")
js := strings.Join(javascript, "\n")

pc := &neosync_benthos.ProcessorConfig{}
if len(mutation) > 0 {
  pc.Mutation = &mutation
}
if len(js) > 0 {
  pc.Javascript = &js
}
```